### ISSUE TITLE
Thoughts on TIC-80 Palette

### ISSUE THREAD

**Issue by sssilver** [nesbox/TIC-80#1300]
Hi,

I am an avid user of TIC-80 as a tool to teach my child basics of programming, and have some thoughts about the TIC-80 palette; feel free to disregard if you find them useless.

So TIC-80 uses the "SWEETIE-16" palette:
![image](https://user-images.githubusercontent.com/619878/104590470-d12d2280-5684-11eb-9062-9827a9d509dd.png)

Compare with PICO-8:
![image](https://user-images.githubusercontent.com/619878/104628963-50851b00-56b2-11eb-837e-ddfcb32be2f0.png)

Here are both palettes stacked on top of each other; TIC-80 is top, Pico-8 is bottom:
![image](https://user-images.githubusercontent.com/619878/104630542-4e23c080-56b4-11eb-8360-49f7eb4cfb72.png)

And here are their color histograms:
TIC-80:
![image](https://user-images.githubusercontent.com/619878/104789726-39782300-57af-11eb-9372-53d26b703143.png)

PICO-8:
![image](https://user-images.githubusercontent.com/619878/104789680-2b2a0700-57af-11eb-9317-aa1d59e33f71.png)

It's obvious that TIC-80's palette is somewhat less vivid and more gloomy. Out of the 16 colors, 11 are in the "gloomy" hue (blue/purple/gray):
![image](https://user-images.githubusercontent.com/619878/104790980-6417ab00-57b2-11eb-8c7e-08152ca9395d.png)

And the "feel" definitely gets through. Out of the 5 kids I have surveyed ranging between 7 and 15 years old, every single one of them preferred the PICO-8 palette. Now of course, TIC-80 is not for children -- it is a serious fantasy computer aimed at adults. However, as adults, we are still psychologically drawn to brighter and warmer visuals; there's enough gloom in the daily lives of most of us.

Moreover, the feel isn't the only issue with the TIC-80 palette. Notably, the color brown is missing, while the palette contains 4 shades of blue. This is extremely inconvenient, because there are many things in life that are, well, brown (namely, anything made of wood).

I appreciate the fact that the palette is customizable, and I am not in any way advocating that TIC-80 should simply _adopt_ PICO-8's palette. I think both palettes could be improved, however currently the TIC-80 palette would benefit a lot more from such an improvement. I think TIC-80 could look a lot more appealing, delightful, and functional out of the box.

**Comment by potato25** [comment]
Like you mentioned before , you can change palette in palette editor . You can see list of palettes here : https://lospec.com/palette-list , and change colors according to your needs . I personally prefer Sweetie-16

**Comment by SuperPupperDoggo** [comment]
Maybe it could be an option in the config cart(perhaps using the overline palette area), and new projects will use that palette(for the scanline one of course).

**Comment by SuperPupperDoggo** [comment]
Maybe default palette could be an option in the config cart(perhaps using the overline palette area), and new projects will use that palette(for the scanline one of course). This would allow everyone to have the default palette they prefer.

**Comment by SuperPupperDoggo** [comment]
Maybe default palette could be an option in the config cart(perhaps using the overline palette area), and new projects will use that palette(for the normal palette area of course). This would allow everyone to have the default palette they prefer.

**Comment by sssilver** [comment]
Orthogonally, it would also be a lot more convenient if the palette could be customized in cartridge code rather than the UI.

**Comment by SuperPupperDoggo** [comment]
it can be: https://github.com/nesbox/TIC-80/wiki/code-examples-and-snippets#pal-function

**Comment by sssilver** [comment]
I guess I meant it in a more declarative/configuration fashion

**Comment by SuperPupperDoggo** [comment]
so like a metadata thing? what format would it be in?

**Comment by sssilver** [comment]
I don't know, a Lua dictionary? JSON? YAML?

**Comment by SuperPupperDoggo** [comment]
I suppose

**Comment by nesbox** [comment]
> Maybe default palette could be an option in the config cart(perhaps using the overline palette area), and new projects will use that palette(for the normal palette area of course). This would allow everyone to have the default palette they prefer.

Try to use `config default lua` command to edit the default TIC cart.

**Comment by borbware** [comment]
it would be nice if the default palette of the game was applied to the console as well, at least in the built version!

**Comment by nesbox** [comment]
It was like that before in the first releases, then we split it into the system palette and game palette.

**Comment by borbware** [comment]
maybe an option could be added to config file where one could choose if the game palette is applied as a system palette on export?

**Comment by Skaruts** [comment]
> it would be nice if the default palette of the game was applied to the console as well, at least in the built version!

A problem with that is that changing the colors in the game's palette will make the system look weird.

**Comment by Skaruts** [comment]
I gave this palette a try many times over time, and I ended up disliking it a lot. It looks good, but it's very hard to use. I'm not a beginner, but when it comes to palettes I'm not much better than any beginner. I don't have any sophisticated knowledge of palettes. I just go by eye and by feeling.

But I still had no issues with the old palette (from 0.70), or with Pico-8's palette. I can easily draw lots of stuff out of the box with that. It had a good enough assortment of colors that made it easy for a noob like me to figure out how to use one shading to complement another. Like using dark red or even pink as shades of wood.

With this palette I always get stuck, being unable to draw certain things. Wood is a great example, because there's indeed no brown, and there are also not enough tones that can look like shades of brown. So every single time I just ended up giving up and launching 0.70 and copied the old palette into my current project.

I suspect this is also not any good for beginners. I know one can switch the default cart's palette, but most people won't know that for a while (I took years to realize that was a thing), and will just go with what's there, and will find it unnecessarily difficult to work with.

**Comment by MypkaMax** [comment]
I made a general purpose 16-colour palette and it's still pending on Lospec. It's supposed to evoke the feeling of PC-98 games, hence the "PC-16" moniker, but now that I found this conversation, I'm wondering if this palette would fit the bill for a vibrant default palette people want in TIC-80. And yes, I know I'm a year late.
https://lospec.com/palette-list/myp-pc-16
